internal/client/cloudfoundry/client.go

```diff
@@ -27,5 +121,479 @@ func NewClient(config *config.Config, logger *gosteno.Logger) (*cfclient.Client,
+		processes, exists := processesPerApp[appGUID]
+		if exists {
+			updatedApp.setV3ProcessData(processes)
+		} else {
+			cfc.logger.Errorf("could not fetch processes info for app guid %s", appGUID)
+		}
+		space, exists := spacesPerGUID[spaceGUID]
+		if exists {
+			updatedApp.setV3SpaceData(space)
+		} else {
+			cfc.logger.Errorf("could not fetch space info for space guid %s", spaceGUID)
+		}
+		orgGUID := updatedApp.OrgGUID
+		org, exists := orgsPerGUID[orgGUID]
+		if exists {
+			updatedApp.setV3OrgData(org)
+		} else {
+			cfc.logger.Errorf("could not fetch org info for org guid %s", orgGUID)
+		}
+		results = append(results, updatedApp)
+	}
+
+	return results, nil
+}
+
+func (cfc *CFClient) getV3Apps() ([]CFApplication, error) {
+	// Query the first page to get the total number of pages.
+	cfapps, pages, err := cfc.getV3AppsByPage(1)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error requesting v3 apps page 1, skipping cache warmup")
+	}
+
+	for page := 2; page <= pages; page++ {
+		pageResults, _, err := cfc.getV3AppsByPage(page)
+		if err != nil {
+			// TODO: previously this just logged the error and did "continue",
+			// I'm not sure what is the implication of returning error instead,
+			// but I think it's the right thing to do
+			return nil, err
+		}
+		cfapps = append(cfapps, pageResults...)
+	}
+
+	return cfapps, nil
+}
+
+func (cfc *CFClient) getV3AppsByPage(page int) ([]CFApplication, int, error) {
+	q := url.Values{}
+	q.Set("per_page", "5000") // 5000 is the max
+	if page > 0 {
+		q.Set("page", strconv.Itoa(page))
+	}
+	r := cfc.client.NewRequest("GET", "/v3/apps?"+q.Encode())
+	resp, err := cfc.client.DoRequest(r)
+	if err != nil {
+		return nil, -1, errors.Wrapf(err, "Error requesting apps page %d", err)
+	}
+	// Read body response
+	defer resp.Body.Close()
+	resBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, -1, errors.Wrapf(err, "Error reading app response for page %d", page)
+	}
+	// Unmarshal body response into v3AppResponse objects
+	var appResp v3AppResponse
+	err = json.Unmarshal(resBody, &appResp)
+	if err != nil {
+		return nil, -1, errors.Wrapf(err, "Error unmarshalling app response for page %d", page)
+	}
+	// Create CFApplication objects
+	appResources := appResp.Resources
+	results := []CFApplication{}
+	for _, app := range appResources {
+		cfapp := CFApplication{}
+		cfapp.setV3AppData(app)
+		results = append(results, cfapp)
+	}
+
+	return results, appResp.Pagination.TotalPages, nil
+}
+
+func (cfc *CFClient) getV3Processes() ([]cfclient.Process, error) {
+	// Query the first page to get the total number of pages.
+	results, pages, err := cfc.getV3ProcessesByPage(1)
+	if err != nil {
+		return nil, errors.Wrap(err, "Error requesting v3 processes page 1, skipping cache warmup")
+	}
+
+	for page := 2; page <= pages; page++ {
+		pageResults, _, err := cfc.getV3ProcessesByPage(page)
+		if err != nil {
+			// TODO: previously this just logged the error and did "continue",
```

Comment:
@zippolyte and the same here, WDYT? (IMO not getting up to 5000 results is a good enough reason to mark the whole operation as failed).

Reply:
Exactly my reasoning on the above comment ;)

Reply:
Done.